### PR TITLE
refactor(policy): provide new contexts and function interfaces for policy engine

### PR DESCRIPTION
**Pull request by ndr-brt**
## What this PR changes/adds

Given that #4535 is really huge, I extracted a smaller PR, that takes care of:
- creating new `PolicyContext` implementation
- redefine policy function and policy validator hierarchy 

## Why it does that

help the review process

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Part of #3511 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: core/control-plane/control-plane-aggregate-services/build.gradle.kts [hunk 1]
```diff
@@ -17,14 +17,15 @@ plugins {
 }
 
 dependencies {
-    implementation(project(":spi:common:validator-spi"))
-    implementation(project(":spi:control-plane:control-plane-spi"))
+    api(project(":spi:common:boot-spi"))
```
Comment (jimmarino) [inline core/control-plane/control-plane-aggregate-services/build.gradle.kts:20]: Why are these exposed as APIs? They are not needed to extend this module.

**Review by jimmarino — COMMENTED** [review]
One comment about the dependencies

File: core/control-plane/control-plane-aggregate-services/build.gradle.kts (hunk 1 above)
Comment (ndr-brt) [reply core/control-plane/control-plane-aggregate-services/build.gradle.kts:20]: must have been some rebase/pr split issue, will remove it

Comment (jimmarino) [reply core/control-plane/control-plane-aggregate-services/build.gradle.kts]: I think a module should only expose another module as an `api` if it is transitively exposed to interact with it, e.g. an SPI interface references a type from the other module.

**Review by jimmarino — APPROVED** [review]

**Review by wolf4ood — APPROVED** [review]

**Pull request merged**